enterprise/backend/src/metabase_enterprise/database_routing/common.clj

```diff
@@ -22,7 +22,7 @@
    (router-db-or-id->destination-db-id @api/*current-user* db-or-id))
   ([user db-or-id]
    (when-let [attr-name (user-attribute db-or-id)]
-     (let [database-name (get (:login_attributes user) attr-name)]
+     (let [database-name (get (api/current-user-attributes) attr-name)]
```

Comment:
I don't think this will have any ill effects in practice because I've noticed we're never calling this function with a user other than `api/*current-user*`, but seems like a footgun to ignore the passed user here.

One option might be to just make sure all uses of this function call it with the 1-arity version (only the `db-or-id` without the user) and then remove the 2-arity version since it doesn't seem to be needed in practice. I think the function is only called in a few places so it shouldn't be that hard (I think I did this for the tenant attributes bits).

Reply:
Yeah almost all the calls to this function were passing in @api/*current-user* it looked like.